compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/InteractionUIView.uikit.kt

```diff
@@ -98,20 +99,22 @@ private val UIGestureRecognizerState.isOngoing: Boolean
 /**
  * Enum class representing the possible hit test result of [InteractionUIViewHitTestResult].
  * This enum is used solely to determine the strategy of touch event delivery and
- * doesn't require any additional information about the hit-tested view.
- *
- * [SELF] - Hit-tested Compose view.
- * [COOPERATIVE_CHILD_VIEW] - Hit-tested child view that is okay with Compose receiving touches and
- * delaying them for the child view.
- * [UNCOOPERATIVE_CHILD_VIEW] - Hit-tested child view that doesn't want to cooperate with
- * Compose and receives touches immediately.
- * [NONE] - Hit test didn't yield any result.
+ * doesn't require any additional information about the hit-tested view itself.
  */
-private enum class InteractionUIViewHitTestResult {
-    SELF,
-    COOPERATIVE_CHILD_VIEW,
-    UNCOOPERATIVE_CHILD_VIEW,
-    NONE
+private sealed interface InteractionUIViewHitTestResult {
+    data object Self : InteractionUIViewHitTestResult
+
+    data object NonCooperativeChildView : InteractionUIViewHitTestResult
+
+    /**
+     * Hit test result is Cooperative child view, that allows a delay of [delay] milliseconds.
+     */
+    class CooperativeChildView(
+        val delay: Int
```

Comment:
as it was renamed to delayMs [here](https://github.com/JetBrains/compose-multiplatform-core/pull/1494#discussion_r1724902731), may be change here will be also nice

Reply:
Makes sense indeed.